File: domain/src/main/java/mixter/domain/identity/UserId.java
package mixter.domain.identity;

public class UserId {
    private String email;

    public UserId(String email) {
        if (email == null || email.trim().isEmpty()) {
            throw new UserEmailCannotBeEmpty();
        }
        this.email = email;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        UserId userId = (UserId) o;

        return !(email != null ? !email.equals(userId.email) : userId.email != null);

    }

    @Override
    public int hashCode() {
        return email != null ? email.hashCode() : 0;
    }

    @Override
    public String toString() {
        return email;
    }
}

File: domain/src/main/scala/mixter/domain/identity/UserId.scala
package mixter.domain.identity

import mixter.domain.AggregateId

case class UserId(email: String) extends AggregateId

object UserId {
  def of(email: String): Option[UserId] =
    Option(email).filter(_.nonEmpty).map(s=>UserId(s))
}
